adafruit_trellism4.py

```diff
@@ -30,14 +30,20 @@
 import neopixel
 import adafruit_matrixkeypad
 
+try:
+    from typing import List, Optional, Tuple, Union
+    from microcontroller import Pin
+except ImportError:
+    pass
+
 __version__ = "0.0.0+auto.0"
 __repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_TrellisM4.git"
 
 
 class _NeoPixelArray:
     """Creates a NeoPixel array for use in the ``TrellisM4Express`` class."""
 
-    def __init__(self, pin, *, width, height, rotation=0):
+    def __init__(self, pin: Pin, *, width: int, height: int, rotation: int = 0) -> None:
```

Comment:
`rotation` can be `int`, but it looks like issues arise in `_calculate_pixel_offset()` if it isn't `Literal[0, 90, 180, 270]`.  Let's add `typing-extensions~=4.0` to `requirements.txt`, import `Literal` from `typing_extensions`, and then type it as such here.